src/qamqpclient_p.h

```diff
@@ -100,6 +101,12 @@ class QAMQP_EXPORT QAmqpClientPrivate : public QAmqpMethodFrameHandler
     QAMQP::Error error;
     QString errorString;
 
+    /*! Exchange objects */
+    QAmqpChannelHash exchanges;
```

Comment:
is there a reason you have two hashes here? QAmqpChannelHash is already built for abstracting the channels themselves, so you can store them all in a single hash.